### ISSUE TITLE
TracerMiddleware: consider supporting Starlettes add_middleware API

### ISSUE THREAD

**Issue by casio** [florimondmanca/ddtrace-asgi#9]
Starlettes(and in extension FastAPIs) signature for adding middleware using the `add_middleware` method only allows forwarding kwargs to the constructor.

So using eg
`app.add_middleware(TraceMiddleware, tracer, service="some-service")` 

will raise
`TypeError: add_middleware() takes 2 positional arguments but 3 were given`

Maybe just have `tracer` become a keyword argument?

**Comment by florimondmanca** [comment]
Hey, thanks for opening this! :)

> Maybe just have tracer become a keyword argument?

Just to confirm — is this currently preventing you of using this with Starlette, or just a suggestion to improve consistency?

Either way, I’d be fine with turning all arguments of the middleware (except app I suppose?) into keyword-only, yup.

**Comment by casio** [comment]
> Hey, thanks for opening this! :)

Hehe, thanks for getting back so quickly! : )

I think I _could_ just use the `app = TraceMiddleware(app, tracer, ...)` syntax.
However, in my current app, I just prefer writing it more explicitly and also as described in eg the Starlette docs [0]:
```
app = FastAPI()
app.add_middleware(SentryAsgiMiddleware)
app.add_middleware(TraceMiddleware, tracer=tracer, service="some-service")
app.add_middleware(CORSMiddleware, allow_origins=[...], ...)
```

`.add_middleware` is just a `Starlette` derived method, providing `app` as the only positional arg plus all kwargs to the given middleware ctor.

[0] https://www.starlette.io/middleware/

**Comment by florimondmanca** [comment]
Released in [0.2.0](https://pypi.org/project/ddtrace-asgi/0.2.0). :-) Thanks!
